### PR TITLE
Improved case-insensitive match to account for special characters.

### PR DESCRIPTION
**Pull request by crisely09**
Related to https://github.com/BlueBrain/nexus-forge/issues/276
Added some changes to detect special characters and account them in the the Filter regex SPARQL query.

### CODE REVIEW

**Comment by lecriste** [comment]
I tested this PR with a few labels and it's working fine, thanks!

File: examples/notebooks/use-cases/ResolvingStrategies.ipynb [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+       "Resource(_last_action=None, _validated=False, _synchronized=False, _store_metadata=None, id='https://bbp.epfl.ch/ontologies/core/ttypes/264_L5_6_NP_CTX', type=['Entity', 'Class'], label='264_L5/6 NP CTX', _inner_sync=False, subClassOf=['https://bbp.epfl.ch/ontologies/core/ttypes/L5_6_NP_CTX', 'https://bbp.epfl.ch/ontologies/core/celltypes/NeuronTranscriptomicType'])"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "forge.resolve('264_L5/6 np CTX', scope='ontology', strategy='EXACT_CASEINSENSITIVE_MATCH')"
```
Comment (crisely09) [inline examples/notebooks/use-cases/ResolvingStrategies.ipynb:124]: This is a very nasty case, and we can resolve it!

File: kgforge/specializations/resolvers/agent_resolver.py [hunk 2]
```diff
@@ -52,9 +53,11 @@ def _resolve(self, text: Union[str, List[str]], target: Optional[str], type: Opt
             family_name_filter = f" FILTER (?familyName = \"{text}\")"
             limit = 1
         elif strategy == strategy.EXACT_CASEINSENSITIVE_MATCH:
-            name_filter = f" FILTER regex(?name, \"^{text}$\", \"i\")"
-            given_name_filter = f" FILTER regex(?givenName, \"^{text}$\", \"i\")"
-            family_name_filter = f" FILTER regex(?familyName, \"^{text}$\", \"i\")"
+            tmp_text = re.sub(r'[-()\"#/@;:<>{}`+=~|.!?,]', "\\\\S", text)
```
Comment (crisely09) [inline kgforge/specializations/resolvers/agent_resolver.py:56]: Here I keep track of the positions of the special characters (assigning a temporal value that I can replace), so in the next line I use `\p{Punct}` in those positions, to pass to the regex filter.

File: kgforge/specializations/resolvers/ontology_resolver.py [hunk 3]
```diff
@@ -48,25 +50,23 @@ def _resolve(self, text: Union[str, List[str]], target: Optional[str], type: Opt
...
 
+        properties = ['label', 'notation', 'prefLabel', 'altLabel']
         if strategy == strategy.EXACT_MATCH:
-            label_filter = f" FILTER (?label = \"{text}\")"
-            notation_filter = f" FILTER (?notation = \"{text}\")"
-            prefLabel_filter = f" FILTER (?prefLabel = \"{text}\")"
-            altLabel_filter = f" FILTER (?altLabel = \"{text}\")"
-            limit = 1
+            regex = False
+            case_insensitive = False
```
Comment (MFSY) [inline kgforge/specializations/resolvers/ontology_resolver.py:56]: Did we lost the limit = 1 for EXACT_MATCH or is it some default value somewhere?

**Review by MFSY — APPROVED** [review]

**Pull request merged**